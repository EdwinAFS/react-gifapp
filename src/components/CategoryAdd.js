import React, { useState } from "react";
import PropTypes from "prop-types";

export const CategoryAdd = ({setCategories}) => {

	const [categoryInput, setCategoryInput] = useState('');

	const handleChangeCategory = (e) => {
		setCategoryInput(e.target.value);
	}

	const handleSubmit = ( e ) => {
		e.preventDefault();

		if( categoryInput.trim().length < 2){
			return;
		}

		setCategories( categories => [categoryInput, ...categories] );
	}

	return (
		<form onSubmit={handleSubmit}>
			<input
				type="text"
				value={categoryInput}
				onChange={handleChangeCategory}
			/>
		</form>
	);
};

CategoryAdd.propTypes = {
	setCategories: PropTypes.func.isRequired
}