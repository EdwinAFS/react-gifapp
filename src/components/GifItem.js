import React from 'react';
import PropTypes from 'prop-types';

export const GifItem = ({ title, url }) => {

	return (
		<div className='card animate__animated animate__fadeInDown'>
			<div className='body'>
				<img src={url} alt={title}/>
			</div>
			<div className='footer'>
				<p>{title}</p>
			</div>
		</div>
	);
}


GifItem.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired
}