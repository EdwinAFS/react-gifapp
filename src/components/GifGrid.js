import React, { Fragment } from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem'; 
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

	const { data: gifs, loading } = useFetchGifs( category );

	return (
		<Fragment>
			<h3>{category}</h3>

			{ loading && <p className='animate__animated animate__lightSpeedInLeft'>Cargando...</p>}
			<div className='grid'>
				{
					gifs.map(gif => {
						return (
							<GifItem key={gif.id} {...gif} />
						);
					})
				}
			</div>

		</Fragment>
	);

}

GifGrid.protoType = {
	category: PropTypes.string.isRequired
}