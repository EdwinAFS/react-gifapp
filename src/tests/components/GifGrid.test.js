import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('Test para el componente GifGrid', () => {

	const category = 'Vikings';

	test('Deberia mostrar el componente GifGrid', ()=>{
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true
		});
		 const wrapper = shallow(<GifGrid category={category}/>);
		 expect(wrapper).toMatchSnapshot();
	});
	
	test('Deberia mostrar los items de las imagen cuando las obtiene', ()=>{

		const gifs = [{
			id: 'ABC',
			url: 'https://localhost/test/test.jpg',
			title: 'test'
		}];

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false
		});

		const wrapper = shallow(<GifGrid category={category}/>);

		const loading = wrapper.find('p');
		const GifItem = wrapper.find('GifItem');

		expect( loading.exists() ).toBe( false );
		expect( GifItem.length ).toBe( gifs.length );
		 
	});



});