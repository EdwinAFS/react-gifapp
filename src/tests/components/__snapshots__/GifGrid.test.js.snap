// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test para el componente GifGrid Deberia mostrar el componente GifGrid 1`] = `
<Fragment>
  <h3>
    Vikings
  </h3>
  <p
    className="animate__animated animate__lightSpeedInLeft"
  >
    Cargando...
  </p>
  <div
    className="grid"
  />
</Fragment>
`;
